mungegithub/mungers/submit-queue.go

```diff
@@ -1165,7 +1172,32 @@ func (sq *SubmitQueue) doGithubE2EAndMerge(obj *github.MungeObject) bool {
 		}
 		glog.Infof("Skipping retest since head and base sha match previous attempt!")
 		atomic.AddInt32(&sq.retestsAvoided, 1)
-	} else if sq.retestPR(obj) {
+	} else {
+		if sq.retestPR(obj) {
+			return true
+		}
+
+		err := obj.Refresh()
+		if err != nil {
+			glog.Errorf("%d: unknown err: %v", *obj.Issue.Number, err)
+			sq.SetMergeStatus(obj, unknown)
+			return true
+		}
+	}
+
+	// We shouldn't merge if it's not valid anymore
+	if !sq.validForMerge(obj) {
+		glog.Errorf("%d: Not mergeable anymore. Do not merge.", *obj.Issue.Number)
```

Comment:
we should probably do sq.setMergeStatus here with the "sha changed", and return false to do a retest.